wdl/downsample.wdl

```diff
@@ -1,305 +1,284 @@
+        ref_fai = ref_fai,
+        ref_dict = ref_dict,
+        original_cram_or_bam_file_read_groups = downSampling_01.read_groups,
+        intervals_exons = intervals_exons,
+        sample_ID = sample_ID,
+        gatk4_jar_override = gatk4_jar_override,
+        gatk_docker = gatk_docker,
+        runtime_attr_random_sample = runtime_attr_random_sample,
+        runtime_attr_realign = runtime_attr_realign,
+        runtime_attr_add_read_group = runtime_attr_add_read_group,
+        runtime_attr_mark_duplicates = runtime_attr_mark_duplicates,
+        runtime_attr_sort_index = runtime_attr_sort_index,
+        runtime_attr_count_coverage = runtime_attr_count_coverage,
+        runtime_attr_collect_counts = runtime_attr_collect_counts
 
-    command <<<
-        NT=$(nproc)
-
-        bwa mem -C \
-                -t ${NT} \
-                ${reference_fasta} \
-                ${downsample_file_1} \
-                ${downsample_file_2} \
-                | samtools view \
-                -bS - > ${bam_downsample_name}
-
-        samtools view \
-                -C \
-                -T ${reference_fasta} \
-                ${bam_downsample_name} \
-                > ${cram_downsample_name}
-    >>>
-
-    runtime {
-        docker: downsample_docker
     }
-}
-
-task sortIndex {
-    
-    input {
-
-        File cram_downsample_file
-        String downsample_docker
+  }
+
+  if (run_downsample_8x) {
+    call ds2.downSampling_02 as downSampling_02_8x {
+      input :
+        fastq_file_1 = downSampling_01.fastq_1,
+        fastq_file_2 = downSampling_01.fastq_2,
+        downsample_docker = downsample_docker,
+        start_depth = start_depth,
+        final_depth = final_depth_8x,
+        seed_override = seed_override,
+        reference_fasta = reference_fasta,
+        ref_amb = ref_amb,
+        ref_ann = ref_ann,
+        ref_bwt = ref_bwt,
+        ref_pac = ref_pac,
+        ref_sa = ref_sa,
+        ref_fai = ref_fai,
+        ref_dict = ref_dict,
+        original_cram_or_bam_file_read_groups = downSampling_01.read_groups,
+        intervals_exons = intervals_exons,
+        sample_ID = sample_ID,
+        gatk4_jar_override = gatk4_jar_override,
+        gatk_docker = gatk_docker,
+        runtime_attr_random_sample = runtime_attr_random_sample,
+        runtime_attr_realign = runtime_attr_realign,
+        runtime_attr_add_read_group = runtime_attr_add_read_group,
+        runtime_attr_mark_duplicates = runtime_attr_mark_duplicates,
+        runtime_attr_sort_index = runtime_attr_sort_index,
+        runtime_attr_count_coverage = runtime_attr_count_coverage,
+        runtime_attr_collect_counts = runtime_attr_collect_counts
     }
-
-    String downsample_file_sorted_name = basename(cram_downsample_file, ".cram") + "_sorted.cram"
-    
-    output {
-        File cram_sorted_file = downsample_file_sorted_name
-        File crai_file = cram_sorted_file + ".crai"
+  }
+
+  if (run_downsample_custom) {
```

Comment:
I would add an additional check to make sure that the `final_depth_custom` input is defined, since that Float is required for this version of the workflow to be called. I see that you added a catch to set it to 1 if it's not provided, which works, but I think it would be better to not spend extra computation on it if it's not set. You should be able to do it like this, or similar:

```suggestion
  if (run_downsample_custom && defined(final_depth_custom)) {
```

You could also just check if that custom depth is defined and get rid of the Boolean, but that's up to you. It's an extra variable, but it provides a nice symmetry with the other run Booleans, so I don't have an issue with it.

```suggestion
  if (defined(final_depth_custom)) {
```